### DISCUSSION TITLE
AMD和intel的CPU性能差距很明显，符合预期嘛

### DISCUSSION THREAD

**Discussion by bendanzzc** [milvus-io/milvus#19203] [Q&A and General discussion]
用相同频率的 intel cpu和AMD cpu 效果差别有百倍，AMD的巨慢，请问符合预期嘛，我看文档里是 推荐intel的cpu，请问有什么说法嘛

**Comment by bendanzzc** [comment]
好像是AVX512的问题

**Comment by yhmo** [comment]
跟使用的指令集有关，不同的CPU对指令集的支持也不相同。

We made an improvement on faiss, use different instructions according to CPU ability:
```
if cpu support AVX512 {
   use AVX512
   print "faiss hook AVX512"
} else if cpu support AVX2 {
   use AVX2
   print "faiss hook AVX2"
} else if cpu support SSE4_2 {
   use SSE4_2
   print "faiss hook SSE4_2"
} else {
   use software computation
   print "faiss hook REF"
}
```
So, you can perform a search request and find "faiss hook xxx" in the log file of querynode, then you will know which kind of instruction it used.
If the log contains "faiss hook REF", it means your CPU cannot support AVX512/AVX2/SSE4_2. The search engine uses software method to do ANN search, which lead to poor performance.
